File: Leetcode/Python/Q4_Median_of_Two_Sorted_Arrays.py
class Solution(object):
    def findMedianSortedArrays(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: float
        """
        # Merge both sorted arrays
        nums = nums1 + nums2
        # Sort the merged array
        nums.sort()

        # Calculate the total length of the merged array
        n = len(nums)

        # If the length is odd, return the middle element
        if n % 2 != 0:
            return float(nums[n // 2])
        else:
            # If the length is even, return the average of the two middle elements
            mid1 = nums[(n // 2) - 1]
            mid2 = nums[n // 2]
            return (mid1 + mid2) / 2.0


        # print(nums)

        

File: Leetcode/Java/Q4_Median_of_Two_Sorted_Arrays.java
class Solution {
    public double findMedianSortedArrays(int[] arr1, int[] arr2) {
        int len1 = arr1.length;
        int len2 = arr2.length;
        int totalLength = len1 + len2;

        // Merge the two sorted arrays
        int[] mergedArray = new int[totalLength];
        int i = 0, j = 0, k = 0;

        while (i < len1 && j < len2) {
            if (arr1[i] < arr2[j]) {
                mergedArray[k++] = arr1[i++];
            } else {
                mergedArray[k++] = arr2[j++];
            }
        }

        // Copy remaining elements from arr1 if any
        while (i < len1) {
            mergedArray[k++] = arr1[i++];
        }

        // Copy remaining elements from arr2 if any
        while (j < len2) {
            mergedArray[k++] = arr2[j++];
        }

        // Find the median from the merged sorted array
        return findMedian(mergedArray);
    }

    // Helper method to find the median of a sorted array
    private double findMedian(int[] arr) {
        int n = arr.length;
        if (n % 2 == 0) {
            // If the length of the array is even, average the two middle elements
            return (arr[n / 2] + arr[n / 2 - 1]) / 2.0;
        } else {
            // If the length of the array is odd, return the middle element
            return arr[n / 2];
        }
    }
}
